src/NuGetForUnity/Editor/NugetHelper.cs

```diff
@@ -1136,32 +1114,16 @@ private static NugetPackage GetOnlinePackage(NugetPackageIdentifier packageId)
         }
 
         /// <summary>
-        ///     Copies the contents of input to output. Doesn't close either stream.
-        /// </summary>
-        private static void CopyStream(Stream input, Stream output)
-        {
-            var buffer = new byte[8 * 1024];
-            int len;
-            while ((len = input.Read(buffer, 0, buffer.Length)) > 0)
-            {
-                output.Write(buffer, 0, len);
-            }
-        }
-
-        /// <summary>
-        ///     Installs the package given by the identifier.  It fetches the appropriate full package from the installed packages, package cache, or package
+        ///     Installs the package given by the identifier. It fetches the appropriate full package from the installed packages, package cache, or package
         ///     sources and installs it.
         /// </summary>
         /// <param name="package">The identifier of the package to install.</param>
         /// <param name="refreshAssets">True to refresh the Unity asset database.  False to ignore the changes (temporarily).</param>
-        /// <param name="isUpdate">True to indicate we're calling method as result of Update and don't want to go through IsAlreadyImportedInEngine</param>
         /// <param name="installDependencies">True to also install all dependencies of the <paramref name="package" />.</param>
-        internal static bool InstallIdentifier(NugetPackageIdentifier package,
-            bool refreshAssets = true,
-            bool isUpdate = false,
-            bool installDependencies = true)
+        /// <returns>True if the package was installed successfully, otherwise false.</returns>
+        internal static bool InstallIdentifier(INugetPackageIdentifier package, bool refreshAssets = true, bool installDependencies = true)
```

Comment:
The `AlreadyImportedLibs` managed by `UnityPreImportedLibraryResolver` should never contain any NuGet package installed by NuGetForUnity, if so there are many possible issues e.g. in the UI the "update" button will not be displayed. 
It explicitly removes them, see:
https://github.com/GlitchEnzo/NuGetForUnity/blob/a37c7b09b4d2bddfc2820e4b8aa461dfe0c9d28d/src/NuGetForUnity/Editor/UnityPreImportedLibraryResolver.cs#L29-L41
Before this PR there was a edge-case that when the `NuGet.config` file was not loaded the DLL's are not excluded. I resolved it. I will also add a `if (IsAlreadyImportedInEngine(package, false))` in the start of `Uninstall` so even if the package is uninstalled before the list is filled Unity can't be tricked. The `isUpdate` solution wouldn't work in all cases e.g. when a dependency also need to be updated.

Reply:
I am not sure you understood me. I am aware of the change you linked above. That code is removing things that are installed by NugetForUnity from the list of all loaded assemblies, but the issue is that it is looking at the file system to see what did NugetForUnity install and in the middle of the update procedure the package will still be in the list of all loaded assemblies and it will not be on the file system so that code you linked will not remove it from the list. Please read my first message carefully again to realize how that happens.

I don’t see how adding that check to Uninstall will help with this issue.

Reply:
The "already imported list" is static, and only filled once.
https://github.com/GlitchEnzo/NuGetForUnity/blob/a37c7b09b4d2bddfc2820e4b8aa461dfe0c9d28d/src/NuGetForUnity/Editor/UnityPreImportedLibraryResolver.cs#L14
So as long as unity doesn't reload the whole AppDomain there is no way for the package to "get into the list" while a package update is performed.

Reply:
Ah, I see. Nice solution then 😀.